posix/clone

```diff
@@ -2,6 +2,19 @@
 
 set -e
 
+# if PLUGIN_COPY_FILE_CONTENT is set, only copy the file content
+if [ "$PLUGIN_COPY_FILE_CONTENT" = "true" ]; then
+  if [ ! -z "$PLUGIN_OUTPUT_FILE_PATHS_CONTENT" ]; then
```

Comment:
these changes already exist at the end of this file, please check

Reply:
I have added description devki, We need this functionality without cloning. Thats why new changes has been added

Reply:
this looks like you don't need git clone at all. Whatever you are doing can be done in a run step as well.

Reply:
@devkimittal We don't want to keep duplicate code at the moment. In most of the case we will use content after cloning the code.

Reply:
I still don't understand the use case of gitclone step if we are not cloning anything, what's the use of connector and all? Why can't we use a run step instead, git clone is supposed to clone.

Reply:
let's create another file, name it copy content or something and call it inside the if statement

Reply:
line 7-14 should be a common function since same code is being used below. Can you please refactor ?